pages/api/socketio-init.ts

```diff
@@ -0,0 +1,58 @@
+import axios from "axios"
+import { BehaviorSubject, defer, from, iif, timer } from "rxjs"
+import { map, switchMapTo } from "rxjs/operators"
+import { Server } from "socket.io"
+
+const { IEX_API_TOKEN: TOKEN } = process.env
+
+const sub = new BehaviorSubject<string>("")
+
+const ioHandler = (_req: any, res: any) => {
+  if (!res.socket.server.io) {
+    console.log("*First use, starting websocket")
+    const io = new Server(res.socket.server)
+
+    io.on("connection", (socket) => {
+      console.log("Client has connected")
+
+      socket.on("message", (symbol) => {
+        sub.next(symbol)
+      })
+
+      sub.subscribe((symbol) => {
+        iif(
+          () => (symbol !== null || symbol !== undefined) && symbol.length != 0,
```

Comment:
can shorten to  condition `symbol?.length > 0`

Reply:
👍